test/e2e/e2e_test.go

```diff
@@ -483,6 +485,100 @@ var _ = ginkgo.Describe("Kueue visibility server", func() {
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, clusterRoleBinding)).Should(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(k8sClient.Delete(ctx, clusterRoleBinding)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should give access to pending workloads in ClusterQueue", func() {
+			ginkgo.By("Verify the ClusterQueue does not exist", func() {
+				_, err := impersonatedVisibilityClient.ClusterQueues().GetPendingWorkloadsSummary(ctx, "non-existent", metav1.GetOptions{})
+				statusErr, ok := err.(*errors.StatusError)
+				gomega.Expect(ok).To(gomega.Equal(true))
+				gomega.Expect(statusErr.ErrStatus.Reason).To(gomega.Equal(metav1.StatusReasonNotFound))
+			})
+		})
+
+		ginkgo.It("Should give access to pending workloads in LocalQueue", func() {
+			ginkgo.By("Verify the LocalQueue does not exist", func() {
+				_, err := impersonatedVisibilityClient.LocalQueues(nsA.Name).GetPendingWorkloadsSummary(ctx, "non-existent", metav1.GetOptions{})
+				statusErr, ok := err.(*errors.StatusError)
+				gomega.Expect(ok).To(gomega.Equal(true))
+				gomega.Expect(statusErr.ErrStatus.Reason).To(gomega.Equal(metav1.StatusReasonNotFound))
+			})
+		})
+	})
+
+	ginkgo.When("A subject is bound to kueue-batch-user-role, but not to kueue-batch-admin-role", func() {
+		var roleBinding *rbacv1.RoleBinding
+
+		ginkgo.BeforeEach(func() {
+			roleBinding = &rbacv1.RoleBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "read-pending-workloads", Namespace: nsA.Name},
+				RoleRef:    rbacv1.RoleRef{APIGroup: rbacv1.GroupName, Kind: "ClusterRole", Name: "kueue-batch-user-role"},
+				Subjects: []rbacv1.Subject{
+					{Name: "default", APIGroup: "", Namespace: "kueue-system", Kind: rbacv1.ServiceAccountKind},
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, roleBinding)).Should(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(k8sClient.Delete(ctx, roleBinding)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should restrict access to pending workloads in ClusterQueue", func() {
+			ginkgo.By("Verify there is a permission error", func() {
+				_, err := impersonatedVisibilityClient.ClusterQueues().GetPendingWorkloadsSummary(ctx, "non-existent", metav1.GetOptions{})
+				statusErr, ok := err.(*errors.StatusError)
+				gomega.Expect(ok).To(gomega.Equal(true))
+				gomega.Expect(statusErr.ErrStatus.Reason).To(gomega.Equal(metav1.StatusReasonForbidden))
+			})
+		})
+
+		ginkgo.It("Should give access to pending workloads in LocalQueue", func() {
+			ginkgo.By("Verify the LocalQueue does not exist", func() {
+				_, err := impersonatedVisibilityClient.LocalQueues(nsA.Name).GetPendingWorkloadsSummary(ctx, "non-existent", metav1.GetOptions{})
+				statusErr, ok := err.(*errors.StatusError)
+				gomega.Expect(ok).To(gomega.Equal(true))
+				gomega.Expect(statusErr.ErrStatus.Reason).To(gomega.Equal(metav1.StatusReasonNotFound))
+			})
+		})
```

Comment:
I would like to have one more scenario, that we get Forbidden error, when attempting to query namespace B. This is important to ensure there is no information leak between namespaces for users.